### PR TITLE
[text analytics] opinion mining sample

### LINKED ISSUE #12478: [text analytics] samples for opinion mining

### PR DESCRIPTION
**Pull request by iscai-msft**
fixes #12478 
need this opinion mining pr to be merged first https://github.com/Azure/azure-sdk-for-python/pull/12542

### CODE REVIEW

File: sdk/textanalytics/azure-ai-textanalytics/samples/README.md [hunk 1]
```diff
@@ -61,6 +61,7 @@ what you can do with the Azure Text Analytics client library.
 
 |**Advanced Sample File Name**|**Description**|
 |----------------|-------------|
+|[sample_analyze_sentiment_with_aspects.py][sample_analyze_sentiment_with_aspects] and [sample_analyze_sentiment_with_aspects_async.py][sample_analyze_sentiment_with_aspects_async]|Analyze sentiment in documents with granular analysis into individual aspects present in a sentence|
```
Comment (kristapratico) [inline sdk/textanalytics/azure-ai-textanalytics/samples/README.md:64]: I wonder if we need to include some note that this sample is only supported in 3.1

File: sdk/textanalytics/azure-ai-textanalytics/samples/async_samples/sample_analyze_sentiment_with_aspects_async.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+
+async def main():
+    sample = AnalyzeSentimentWithAspectsSampleAsync()
+    await sample.analyze_sentiment_with_aspects()
+
+
+if __name__ == '__main__':
+    loop = asyncio.get_event_loop()
+    loop.run_until_complete(main())
```
Comment (kristapratico) [inline sdk/textanalytics/azure-ai-textanalytics/samples/async_samples/sample_analyze_sentiment_with_aspects_async.py:118]: nit: newline

File: sdk/textanalytics/azure-ai-textanalytics/samples/README.md (hunk 1 above)
Comment (iscai-msft) [reply sdk/textanalytics/azure-ai-textanalytics/samples/README.md:64]: good point, will add

File: sdk/textanalytics/azure-ai-textanalytics/samples/sample_analyze_sentiment_with_opinion_mining.py [hunk 3]
```diff
@@ -57,46 +57,47 @@ def analyze_sentiment_with_aspects(self):
             "I had a great unobstructed view of the Microsoft campus"
```
Comment (maririos) [inline sdk/textanalytics/azure-ai-textanalytics/samples/sample_analyze_sentiment_with_opinion_mining.py:57]: I can't comment but considering changing class name too

File: sdk/textanalytics/azure-ai-textanalytics/samples/sample_analyze_sentiment_with_opinion_mining.py [hunk 4]
```diff
@@ -57,46 +57,47 @@ def analyze_sentiment_with_aspects(self):
...
+                        mixed_mined_opinions.append(mined_opinion)
                     else:
-                        negative_aspects.append(aspect)
+                        negative_mined_opinions.append(mined_opinion)
 
-        print("\n\nLet's look at the {} positive aspects of this hotel".format(len(positive_aspects)))
-        for aspect in positive_aspects:
-            print("...Reviewers have the following opinions for the overall positive '{}' feature of the hotel".format(aspect.text))
-            for opinion in aspect.opinions:
+        print("\n\nLet's look at the {} positive opinions of this hotel".format(len(positive_mined_opinions)))
```
Comment (maririos) [inline sdk/textanalytics/azure-ai-textanalytics/samples/sample_analyze_sentiment_with_opinion_mining.py:85]: aspects?

Comment (maririos) [reply sdk/textanalytics/azure-ai-textanalytics/samples/sample_analyze_sentiment_with_opinion_mining.py]: comments?

File: sdk/textanalytics/azure-ai-textanalytics/samples/README.md [hunk 5]
```diff
@@ -80,6 +81,8 @@ what you can do with the Azure Text Analytics client library.
 [get_detailed_diagnostics_information_async]: https://github.com/Azure/azure-sdk-for-python/tree/master/sdk/textanalytics/azure-ai-textanalytics/samples/async_samples/sample_get_detailed_diagnostics_information_async.py
 [sample_alternative_document_input]: https://github.com/Azure/azure-sdk-for-python/tree/master/sdk/textanalytics/azure-ai-textanalytics/samples/sample_alternative_document_input.py
 [sample_alternative_document_input_async]: https://github.com/Azure/azure-sdk-for-python/tree/master/sdk/textanalytics/azure-ai-textanalytics/samples/async_samples/sample_alternative_document_input_async.py
+[sample_analyze_sentiment_with_aspects]: https://github.com/Azure/azure-sdk-for-python/tree/master/sdk/textanalytics/azure-ai-textanalytics/samples/sample_analyze_sentiment_with_aspects.py
+[sample_analyze_sentiment_with_aspects_async]: https://github.com/Azure/azure-sdk-for-python/tree/master/sdk/textanalytics/azure-ai-textanalytics/samples/async_samples/sample_analyze_sentiment_with_aspects_async.py
```
Comment (kristapratico) [inline sdk/textanalytics/azure-ai-textanalytics/samples/README.md:85]: this file needs update

File: sdk/textanalytics/azure-ai-textanalytics/samples/async_samples/sample_analyze_sentiment_with_opinion_mining_async.py [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+    Set the environment variables with your own values before running the sample:
+    1) AZURE_TEXT_ANALYTICS_ENDPOINT - the endpoint to your Cognitive Services resource.
+    2) AZURE_TEXT_ANALYTICS_KEY - your Text Analytics subscription key
+"""
+
+import os
+import asyncio
+
+
+class AnalyzeSentimentWithAspectsSampleAsync(object):
```
Comment (kristapratico) [inline sdk/textanalytics/azure-ai-textanalytics/samples/async_samples/sample_analyze_sentiment_with_opinion_mining_async.py:33]: AnalyzeSentimentWithOpinionMiningSampleAsync

File: sdk/textanalytics/azure-ai-textanalytics/samples/async_samples/sample_analyze_sentiment_with_opinion_mining_async.py [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+    async def analyze_sentiment_with_aspects(self):
```
Comment (kristapratico) [inline sdk/textanalytics/azure-ai-textanalytics/samples/async_samples/sample_analyze_sentiment_with_opinion_mining_async.py:34]: sample_analyze_sentiment_with_opinion_mining_async

**Pull request merged**